sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -282,316 +479,363 @@ q.parallel_for(nd_range<2>(G, L), [=](nd_item<2> item)
+implementation supports. In validation mode, where the user does
+provide sizes, this is the same value N that the user provides if N is
+supported by the implementation
+|`K`| when no sizes are provided by the user, indicates the suggested
+default size for K; usually this corresponds to the maximum size the
+implementation supports. In validation mode, where the user does
+provide sizes, this is the same value K that the user provides if K is
+supported by the implementation
+|`template <typename Group, layout Layout> +
+using joint_matrix_a;`| type alias for `joint_matrix` for matrix A
+|`template <typename Group, layout Layout> +
+using joint_matrix_b;`| type alias for `joint_matrix` for matrix B
+|`template <typename Group> +
+using joint_matrix_c;`| type alias for `joint_matrix` for the input matrix accumulator
+|`template <typename Group> +
+using joint_matrix_d;`| type alias for `joint_matrix` for the output matrix accumulator
 |======================
 
-
-
 ```c++
 namespace sycl::ext::oneapi::experimental::matrix {
-template<tpu u, typename Ta=void, typename Tb=void, typename Tc=void, int sM=0, int sN=0, int sK=0>
-struct tpu_params;
-
-// Validation form: Valid or not
-// Specialization when both types and sizes are given
-template <typename Ta, typename Tb, typename Tc, int sM, int sN, int sK, layout>
-struct tpu_params<
-    tpu::amx, Ta, Tb, Tc, sM, sN, sK,
-    typename std::enable_if<(
-        !std::is_same_v<Ta, void> && !std::is_same_v<Tb, void> &&
-        !std::is_same_v<Tc, void> && sM != 0 && sN != 0 && sK != 0)>::type> {
-  // Validate that parameters are supported
-  static_assert(
-      (sM == 0 && sN == 0 && sK == 0) ||
-          (is_combination_valid_amx<Ta, Tb, Tc>(sM, sN, sK)),
-      "Invalid parameters for Intel AMX, query valid types and maximum sizes "
-      "using: "
-      "tpu_params<tpu::amx> myparams; and then check out myparams.combinations array");
-
-
-  using type_a = Ta; // this type alias is not available in the current implementation 
-  using type_b = Tb; // this type alias is not available in the current implementation
-  using type_accumulator = Tc; // this type alias is not available in the current implementation
-
-  // if combination is valid, construct the matrices
-
-  static constexpr std::size_t M = (sM != 0) ? sM : 16;
-  static constexpr std::size_t N = (sN != 0) ? sN : 16;
-  static constexpr std::size_t K =
-      (sK != 0) ? sK : ((sizeof(Ta) == 1) ? 64 : 32);
-
-  template <typename Group, layout LayoutA>
-  using joint_matrix_a = joint_matrix<Group, Ta, use::a, defaultM, defaultK, LayoutA>;
-  template <typename Group, layout LayoutB>
-  using joint_matrix_b = joint_matrix<Group, Tb, use::b, defaultK, defaultN, LayoutB>;
+
+// This is the validation form, when all template parameters are
+// specified.
+template<architecture Arch, typename Ta, typename Tb, typename Tc,
+typename Td, size_t sM, size_t sN, size_t sK>
```

Comment:
```suggestion
template<architecture Arch, typename Ta, typename Tb, typename Tc,
typename Td, size_t sM=0, size_t sN=0, size_t sK=0>
```

You need these default values here if you want the application to be able to use the "default values form" without explicitly providing `0`'s for these template parameters.

Reply:
if sM,sN,sK values are zero, M,N,K return also zero, and it will generate an assertion in this case as zeros is not supported by any hardware. So in the validation form, sM,sN,sK are required.
Are you saying we cannot use the "default values form" specialization of the query if we don't specify zeros here?

Reply:
I think you are talking about the primary definition. I will add defaults there.
This one is a specialization and should not have the defaults.

Reply:
I think the "validation form" is the primary template definition.  In any case ... yes, the default values need to be specified on the primary template definition (not the specialization).